packages/utils/src/promise.ts

```diff
@@ -120,18 +119,17 @@ export async function resolveOrRacePromises<T extends NonEmptyArray<PromiseLike<
 
     return promiseResults;
   } catch (err) {
+    if (err instanceof ErrorAborted) {
+      return promiseResults;
+    }
```

Comment:
If we want to handle abort error in this function, it should be added to both blocks. Consider the scenario where none resolved during `Promise.allSettled` block and they started in `Promise.any` race. As soon the engine is resolved our logic may try to abort builder.